src/plugins/messageClickActions/index.ts

```diff
@@ -46,44 +48,79 @@ const settings = definePluginSettings({
         description: "Enable double click to reply",
         default: true
     },
+    enableCtrlShiftClickToPinUnpin: {
+        type: OptionType.BOOLEAN,
+        description: "Enable Ctrl+Shift click to pin/unpin messages",
+        default: true
+    },
     requireModifier: {
         type: OptionType.BOOLEAN,
         description: "Only do double click actions when shift/ctrl is held",
         default: false
+    },
+    sendUnpinNotification: {
+        type: OptionType.BOOLEAN,
+        description: "Send a bot message when a message is unpinned",
+        default: false
     }
 });
 
+async function pinMessage(channel: any, message: any): Promise<void> {
+    if (!pinModule) return;
+    await pinModule.pinMessage(channel, message.id);
+}
+
+async function unpinMessage(channel: any, message: any): Promise<void> {
+    if (!pinModule) return;
```

Comment:
1. prefer typing strictly instead of abusing `any`
2. do not explicitly type stuff (in this case return type) if it can be inferred
3. null checking pinModule here will just make debugging harder. also `find*Lazy()` never returns null / undefined so `!pinModule` is always false
 
```suggestion
async function pinMessage(channel: Channel, message: Message) {
    await pinModule.pinMessage(channel, message.id);
}

async function unpinMessage(channel: Channel, message: Message) {
```